python/spacewalk/satellite_tools/repo_plugins/yum_src.py

```diff
@@ -683,11 +684,20 @@ def _prep_zypp_repo_url(self, url, uln_repo):
         if self.sslclientkey:
             query_params['ssl_clientkey'] = self.sslclientkey
         new_query = unquote(urlencode(query_params, doseq=True))
-        if self.authtoken or uln_repo:
-            ret_url = "{0}&{1}".format(url, new_query)
-        else:
-            ret_url = "{0}?{1}".format(url, new_query) if new_query else url
-        return ret_url
+        # urlparse cannot handle uln urls, so we need to keep this check
+        if uln_repo:
+            return "{0}&{1}".format(url, new_query)
+        parsed_url = urlparse(url)
+        query = parsed_url.query
+        final_query = "&".join(filter(None, [query, new_query]))
```

Comment:
nitpick: I'd prefer a more verbose way than `filter(None, iterable)`, because it is a special case for `filter(function, iterable)` that I had to look up to understand. With list comprehensions, `filter` is not used that much anymore and not as intuitive.

Reply:
Would something like `[q in [query, new_query] if q != None]` be better?

Reply:
I tried it out and it seems the behavior is different with list comprehensions. If there is no query already present in the url it still puts a `&` at the start. Filter doesn't do that and that is also the desired behavior

Reply:
That's because `filter(None, iterable)` is **special** and exactly the reason I don't like it.
The equivalent list comprehension is `[ q for q in [query, new_query] if q]`:
```python
>>> list(filter(None, ['', "foo"]))
['foo']
>>> [x for x in ['', "foo"] if x]
['foo']
```